src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_CommitUniqueCompletionListItem.cs

```diff
@@ -73,17 +74,17 @@ private void CommitUniqueCompletionListItemAsynchronously()
             // We're kicking off async work.  Track this with an async token for test purposes.
             var token = ((IController<Model>)this).BeginAsyncOperation(nameof(CommitUniqueCompletionListItemAsynchronously));
 
-            var task = currentTask.ContinueWith(t =>
+            var task = currentTask.ContinueWith(async t =>
             {
-                this.AssertIsForeground();
+                await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(alwaysYield: true);
 
                 if (this.sessionOpt == currentSession &&
                     this.sessionOpt.Computation.ModelTask == currentTask)
                 {
                     // Nothing happened between when we were invoked and now.
                     CommitIfUnique(t.Result);
                 }
-            }, CancellationToken.None, TaskContinuationOptions.OnlyOnRanToCompletion, ForegroundTaskScheduler);
+            }, CancellationToken.None, TaskContinuationOptions.OnlyOnRanToCompletion | TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default).Unwrap();
```

Comment:
why ExecuteSynchronously?

Reply:
➡️ This pattern is the subject of the ongoing discussion in https://github.com/dotnet/roslyn/pull/28781#discussion_r208378244.